core/node/provider.go

```diff
@@ -2,50 +2,129 @@ package node
 
 import (
 	"context"
+	"fmt"
 	"time"
 
-	"github.com/libp2p/go-libp2p-routing"
 	"go.uber.org/fx"
 
+	"github.com/ipfs/go-ipfs-config"
 	"github.com/ipfs/go-ipfs/core/node/helpers"
 	"github.com/ipfs/go-ipfs/provider"
+	q "github.com/ipfs/go-ipfs/provider/queue"
+	"github.com/ipfs/go-ipfs/provider/simple"
 	"github.com/ipfs/go-ipfs/repo"
-	"github.com/ipfs/go-ipfs/reprovide"
+	"github.com/libp2p/go-libp2p-routing"
 )
 
 const kReprovideFrequency = time.Hour * 12
 
+// SIMPLE
+
 // ProviderQueue creates new datastore backed provider queue
-func ProviderQueue(mctx helpers.MetricsCtx, lc fx.Lifecycle, repo repo.Repo) (*provider.Queue, error) {
-	return provider.NewQueue(helpers.LifecycleCtx(mctx, lc), "provider-v1", repo.Datastore())
+func ProviderQueue(mctx helpers.MetricsCtx, lc fx.Lifecycle, repo repo.Repo) (*q.Queue, error) {
+	return q.NewQueue(helpers.LifecycleCtx(mctx, lc), "provider-v1", repo.Datastore())
 }
 
-// ProviderCtor creates new record provider
-func ProviderCtor(mctx helpers.MetricsCtx, lc fx.Lifecycle, queue *provider.Queue, rt routing.IpfsRouting) provider.Provider {
-	p := provider.NewProvider(helpers.LifecycleCtx(mctx, lc), queue, rt)
+// SimpleProviderCtor creates new record provider
+func SimpleProviderCtor(mctx helpers.MetricsCtx, lc fx.Lifecycle, queue *q.Queue, rt routing.IpfsRouting) provider.Provider {
+	return simple.NewProvider(helpers.LifecycleCtx(mctx, lc), queue, rt)
+}
+
+// SimpleReproviderCtor creates new reprovider
+func SimpleReproviderCtor(reproviderInterval time.Duration) func(helpers.MetricsCtx, fx.Lifecycle, routing.IpfsRouting, simple.KeyChanFunc) (provider.Reprovider, error) {
+	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, rt routing.IpfsRouting, keyProvider simple.KeyChanFunc) (provider.Reprovider, error) {
+		return simple.NewReprovider(helpers.LifecycleCtx(mctx, lc), reproviderInterval, rt, keyProvider), nil
+	}
+}
 
+// SimpleProviderSysCtor creates new provider system
+func SimpleProviderSysCtor(lc fx.Lifecycle, p provider.Provider, r provider.Reprovider) provider.System {
+	sys := provider.NewSystem(p, r)
 	lc.Append(fx.Hook{
 		OnStart: func(ctx context.Context) error {
-			p.Run()
+			sys.Run()
 			return nil
 		},
 		OnStop: func(ctx context.Context) error {
-			return p.Close()
+			return sys.Close()
 		},
 	})
+	return sys
+}
+
+// SimpleOfflineProviderSysCtor creates a new offline provider system
+func SimpleOfflineProviderSysCtor(p provider.Provider, r provider.Reprovider) provider.System {
```

Comment:
We don't really need to wrap this as can just call `fx.Provide(provider.NewSystem)`

Reply:
I'd like to leave these right now as the names communicate my intent and in the next PR I'm going to make better use of this break up. I will definitely pare down what's not needed from there. Let me know if you object to this.